pkg/processor/runtime/rpc/socket.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2024 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rpc
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"io"
+	"net"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/common/status"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+)
+
+type result struct {
+	StatusCode   int                    `json:"status_code"`
+	ContentType  string                 `json:"content_type"`
+	Body         string                 `json:"body"`
+	BodyEncoding string                 `json:"body_encoding"`
+	Headers      map[string]interface{} `json:"headers"`
+	EventId      string                 `json:"event_id"`
+
+	DecodedBody []byte
+	err         error
+}
+
+type batchedResults struct {
+	results []*result
+	err     error
+}
+
+func newBatchedResults() *batchedResults {
+	return &batchedResults{results: make([]*result, 0)}
+}
+
+type socketConnection struct {
+	conn     net.Conn
+	listener net.Listener
+	address  string
+}
+
+type AbstractSocket struct {
+	*socketConnection
+	Logger     logger.Logger
+	runtime    *AbstractRuntime
+	encoder    EventEncoder
+	cancelChan chan struct{}
+	//status     status.Status
+}
+
+type ControlMessageSocket struct {
+	*AbstractSocket
+}
+
+func NewControlMessageSocket(logger logger.Logger, socketConnection *socketConnection, runtime *AbstractRuntime) *ControlMessageSocket {
+	abstractSocket := &AbstractSocket{
+		socketConnection: socketConnection,
+		Logger:           logger,
+		runtime:          runtime,
+		cancelChan:       make(chan struct{}, 1),
+	}
+	return &ControlMessageSocket{AbstractSocket: abstractSocket}
```

Comment:
I'd pass a `parentLogger` and call `GetChild` here instead of in the caller function.